src/VisibilityDetectingView.tsx

```diff
@@ -1,35 +1,33 @@
-// @flow
-
 import { Dimensions, UIManager, View } from 'react-native';
 import React, { PureComponent } from 'react';
 import type { ElementRef } from 'react';
 import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';
 
 const TICK_INTERVAL = 100;
 
-type Props = {|
-  children: any,
-  onInvisible: ?() => void,
-  onVisible: ?() => void,
-  style: ?ViewStyleProp,
-|};
+type Props = {
+  children: any;
+  onInvisible: () => void | null | undefined;
+  onVisible: () => void | null | undefined;
```

Comment:
Do these actually need to be able to return `null` or `undefined`? `on*` event handlers are pretty much always `void`

Reply:
No, but it makes no difference in practice. Returning `void` alone would work, but we're leaving these sort of slightly-overly-broad things for a later cleanup step.